packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { isUndefined } from 'lodash';
+import React from 'react';
+import { isEmpty } from '../../utils';
+import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
+import { CheckboxHintText } from './CheckboxHintText';
+import { CheckboxIcon } from './CheckboxIcon';
+import { CheckboxInput } from './CheckboxInput';
+import { CheckboxLabel } from './CheckboxLabel';
+import { CheckboxLabelText } from './CheckboxLabelText';
+import { useCheckbox } from './useCheckbox';
+import { Wrapper } from './Wrapper';
+
+type CheckboxProps = {
+  /**
+   * If `true`, sets the checkbox will be checked.
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Checkbox` changes.
+   */
+  onChange?: (isChecked: boolean) => void;
+  /**
+   * Sets the label text of the checkbox
+   */
+  children: string;
+  /**
+   * Help text of the checkbox
+   */
+  helpText?: string;
+  /**
+   * Error text of the checkbox
+   *
+   * Renders when `validationState` is set to 'error'
+   *
+   * Overrides helpText
+   */
+  errorText?: string;
+  /**
+   * If `true`, the checkbox will be indeterminate.
+   * This does not modify the isChecked property.
+   *
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * The name of the input field in a checkbox
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the checkbox input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
```

Comment:
from the [docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox):
>If the value attribute was omitted, the default value for the checkbox is on

Reply:
Yeah that is fine. We don't care what browser does. Consumer will handle passing the `value` prop if they need it. 

> checkbox is on

And when they say the "checkbox is on" it's doesn't mean the checkbox's state will be "checked"